From mboxrd@z Thu Jan  1 00:00:00 1970
Return-Path: <linux-kernel-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 95BADC433EF
	for <linux-kernel@archiver.kernel.org>; Tue,  9 Nov 2021 09:48:43 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id 7D417611F0
	for <linux-kernel@archiver.kernel.org>; Tue,  9 Nov 2021 09:48:43 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S244215AbhKIJv2 (ORCPT <rfc822;linux-kernel@archiver.kernel.org>);
        Tue, 9 Nov 2021 04:51:28 -0500
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:49182 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S243626AbhKIJvU (ORCPT
        <rfc822;linux-kernel@vger.kernel.org>);
        Tue, 9 Nov 2021 04:51:20 -0500
Received: from mail-pf1-x435.google.com (mail-pf1-x435.google.com [IPv6:2607:f8b0:4864:20::435])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id 8B690C061764
        for <linux-kernel@vger.kernel.org>; Tue,  9 Nov 2021 01:48:34 -0800 (PST)
Received: by mail-pf1-x435.google.com with SMTP id y5so8695657pfb.4
        for <linux-kernel@vger.kernel.org>; Tue, 09 Nov 2021 01:48:34 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=sifive.com; s=google;
        h=from:to:subject:date:message-id:in-reply-to:references:mime-version
         :content-transfer-encoding;
        bh=/TnOUl4F123mnvANSe6fIkO63Rk+15kCEettwhNh/z0=;
        b=c+oKiP6aTAXv3kd484eeW4UnfjRGvcHzqaDG3PnqjWgSmokMzhdSoGEqb2RhYVnnVZ
         R2HqxI+1/Q5cHL+qudUrM83P+Vbjav0mg3EldzK514G/Fhqs1LTAJYjoz6BTcY1/f9XH
         gBjCEJ/o8XOwUOdqX6a1rLsCXHK94yYEe3h5cDWi8cVkmnZjKp6fGihl68fWt+E7/qiE
         Q9uJFo06TIPrrjBouzLD49IEzfNwE9db8WFTzrdLWI9bjjgiYgsHb1klYa+JHkTsp3YJ
         DzNyyPVDb7nlRG4dMgbQK/LcbTelCcZnbMeS7i5c1NhSZNzKMmjuGviJ18U11YFL29AD
         UW6Q==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20210112;
        h=x-gm-message-state:from:to:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=/TnOUl4F123mnvANSe6fIkO63Rk+15kCEettwhNh/z0=;
        b=NY0JJlcfxc8E0rVRSxOUpZXaTrZ4uAzOzg6nWTRwlQpUoomP9yV7OWd3qGC2bRrmmV
         JNMhJKqexj9G5Lyaa9So/niLvaQUqC/zoOtPF9Knhp4N2FdA6E5d2Bk85ZxZefr364cg
         HBgvozniTPUPvqVtAocMI5chwBDcm7p5LbTq7+GPhhOdHSSp0Wz0oShcihv6VYAjcvyl
         0UBgR3UMTx5wHfhfgAYioZCFRZ7HqYvf5h1QX9steTH332vzbRFbD68LJxe7+kMH8YqE
         llqbYbnVXH02v4WyDJ58QjQE8OuykCxA9iDW/NIJlUAcoSehlkO4cozw7bqxiNaREEzt
         rTxw==
X-Gm-Message-State: AOAM531HPObMpjRnaEsWTc/I38FFWw7PE5l/1jveaLafjTkMhheqsN6P
        /gtoZjIxBSXQoA90Ph90ABS30Q==
X-Google-Smtp-Source: ABdhPJxslxZmgp6HGLXrNO9ENm+ukRudAEyD3TUYAMc2MiU5skbn5f8H5pd83qLPoeINHQerjDYasg==
X-Received: by 2002:a05:6a00:1c65:b0:49f:d8d0:c5d9 with SMTP id s37-20020a056a001c6500b0049fd8d0c5d9mr6568239pfw.72.1636451314045;
        Tue, 09 Nov 2021 01:48:34 -0800 (PST)
Received: from hsinchu16.internal.sifive.com (59-124-168-89.hinet-ip.hinet.net. [59.124.168.89])
        by smtp.gmail.com with ESMTPSA id cv1sm1940511pjb.48.2021.11.09.01.48.32
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 09 Nov 2021 01:48:33 -0800 (PST)
From:   Greentime Hu <greentime.hu@sifive.com>
To:     palmer@dabbelt.com, paul.walmsley@sifive.com,
        linux-riscv@lists.infradead.org, linux-kernel@vger.kernel.org,
        aou@eecs.berkeley.edu
Subject: [PATCH v9 01/17] riscv: Separate patch for cflags and aflags
Date:   Tue,  9 Nov 2021 17:48:13 +0800
Message-Id: <f76b0312113bf960e114724cb578a060b3f101ab.1636362169.git.greentime.hu@sifive.com>
X-Mailer: git-send-email 2.31.1
In-Reply-To: <cover.1636362169.git.greentime.hu@sifive.com>
References: <cover.1636362169.git.greentime.hu@sifive.com>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
Precedence: bulk
List-ID: <linux-kernel.vger.kernel.org>
X-Mailing-List: linux-kernel@vger.kernel.org

From: Guo Ren <ren_guo@c-sky.com>

Use "subst fd" in Makefile is a hack way and it's not convenient
to add new ISA feature. Just separate them into riscv-march-cflags
and riscv-march-aflags.

Signed-off-by: Guo Ren <ren_guo@c-sky.com>
Signed-off-by: Guo Ren <guoren@linux.alibaba.com>
Signed-off-by: Greentime Hu <greentime.hu@sifive.com>
---
 arch/riscv/Makefile | 18 ++++++++++++------
 1 file changed, 12 insertions(+), 6 deletions(-)

diff --git a/arch/riscv/Makefile b/arch/riscv/Makefile
index 0eb4568fbd29..37de70df4fae 100644
--- a/arch/riscv/Makefile
+++ b/arch/riscv/Makefile
@@ -48,12 +48,18 @@ endif
 endif
 
 # ISA string setting
-riscv-march-$(CONFIG_ARCH_RV32I)	:= rv32ima
-riscv-march-$(CONFIG_ARCH_RV64I)	:= rv64ima
-riscv-march-$(CONFIG_FPU)		:= $(riscv-march-y)fd
-riscv-march-$(CONFIG_RISCV_ISA_C)	:= $(riscv-march-y)c
-KBUILD_CFLAGS += -march=$(subst fd,,$(riscv-march-y))
-KBUILD_AFLAGS += -march=$(riscv-march-y)
+riscv-march-cflags-$(CONFIG_ARCH_RV32I)		:= rv32ima
+riscv-march-cflags-$(CONFIG_ARCH_RV64I)		:= rv64ima
+riscv-march-$(CONFIG_FPU)			:= $(riscv-march-y)fd
+riscv-march-cflags-$(CONFIG_RISCV_ISA_C)	:= $(riscv-march-cflags-y)c
+
+riscv-march-aflags-$(CONFIG_ARCH_RV32I)		:= rv32ima
+riscv-march-aflags-$(CONFIG_ARCH_RV64I)		:= rv64ima
+riscv-march-aflags-$(CONFIG_FPU)		:= $(riscv-march-aflags-y)fd
+riscv-march-aflags-$(CONFIG_RISCV_ISA_C)	:= $(riscv-march-aflags-y)c
+
+KBUILD_CFLAGS += -march=$(riscv-march-cflags-y)
+KBUILD_AFLAGS += -march=$(riscv-march-aflags-y)
 
 KBUILD_CFLAGS += -mno-save-restore
 KBUILD_CFLAGS += -DCONFIG_PAGE_OFFSET=$(CONFIG_PAGE_OFFSET)
-- 
2.31.1


